src/OpenTelemetry.Api/Trace/ActivityExtensions.cs

```diff
@@ -41,10 +41,10 @@ public static void SetStatus(this Activity activity, Status status)
         {
             Debug.Assert(activity != null, "Activity should not be null");
 
-            activity.SetTag(SpanAttributeConstants.StatusCodeKey, SpanHelper.GetCachedCanonicalCodeString(status.CanonicalCode));
+            activity?.SetTag(SpanAttributeConstants.StatusCodeKey, SpanHelper.GetCachedCanonicalCodeString(status.CanonicalCode));
```

Comment:
If you do this, you can remove the Debug.Assert above because the method is now null-safe. But the reason it had the Assert is probably for perf. If it was null in Debug it would error, but in Release, no clocks wasted performing null checks. So the other thing we could do here to preserve the perf is just suppress the warning on this method.

Same comment for the method below.

Reply:
just updated!